### PR TITLE
Make closest not require document.body

### PR DESCRIPTION
**Pull request by antidis**
Allows us to keep our JS in `<HEAD>`.

### CODE REVIEW

File: addon/-private/utils/element/closest.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-VENDOR_MATCH_FNS.some((fn) => {
-  if (typeof document.body[fn] === 'function') {
-    ELEMENT_MATCH_FN = fn;
-    return true;
-  }
-  return false;
-});
+function setElementMatchFn(el) {
+  VENDOR_MATCH_FNS.forEach((fn) => {
+    if ((ELEMENT_MATCH_FN === undefined) && (typeof el[fn] === 'function')) {
```
Comment (antidis) [inline addon/-private/utils/element/closest.js]: Prefers the first match.

File: addon/-private/utils/element/closest.js [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
+      ELEMENT_MATCH_FN = fn;
+    }
+  });
+}
 
 export default function closest(el, selector) {
+  if (ELEMENT_MATCH_FN === undefined) {
+    setElementMatchFn(el);
+  }
```
Comment (antidis) [inline addon/-private/utils/element/closest.js]: By basing it off `el`, and by placing it here, we don't run our check until we actually use the function the first time

There is a slight slowdown here when the function is called, but it should be marginal on the second and subsequent calls.

Comment (antidis) [inline addon/-private/utils/element/closest.js]: And this function is only used once in the component anyway, so there should be effectively no slowdown.

**Comment by antidis** [comment]
cc @runspired @gavinjoyce

**Comment by antidis** [comment]
Tests are failing in CI, but the same tests fail for me in `master` locally right now.

**Comment by antidis** [comment]
Looks like master broke here: https://travis-ci.org/html-next/vertical-collection/builds/226719240

**Comment by runspired** [comment]
Accidentally closed this via the commit.

**Pull request merged**